tests/installation/partitioning_firstdisk.pm

```diff
@@ -17,78 +17,9 @@ use warnings;
 use base "y2logsstep";
 use testapi;
 use version_utils 'is_storage_ng';
-
-sub take_first_disk_storage_ng {
-    return unless is_storage_ng;
-    send_key $cmd{guidedsetup};    # select guided setup
-    assert_screen 'select-hard-disks';
-    # It's not always the case that SUT has 2 drives, for ipmi it's changing
-    # So making it flexible, still assert the screen if want to verify explicitly
-    assert_screen [qw(existing-partitions hard-disk-dev-sdb-selected)];
-    if (match_has_tag 'hard-disk-dev-sdb-selected' || get_var('SELECT_FIRST_DISK')) {
-        if (check_var('VIDEOMODE', 'text')) {
-            if (match_has_tag 'hotkey_d') {
-                send_key 'alt-d';
-            }
-            elsif (match_has_tag 'hotkey_e') {
-                send_key 'alt-e';
-            }
-            else {
-                die 'Needle needs tag \'hotkey_d\' or \'hotkey_e\'';
-            }
-        }
-        else {
-            assert_and_click 'hard-disk-dev-sdb-selected';    # Unselect second drive
-        }
-        assert_screen 'select-hard-disks-one-selected';
-        send_key $cmd{next};
-    }
-
-    # If drive is not formatted, we have select hard disks page
-    # On ipmi we always have unformatted drive
-    if (get_var('ISO_IN_EXTERNAL_DRIVE') || check_var('BACKEND', 'ipmi')) {
-        assert_screen 'existing-partitions';
-        send_key $cmd{next};
-    }
-    assert_screen 'partition-scheme';
-    send_key $cmd{next};
-    # select btrfs file system
-    if (check_var('VIDEOMODE', 'text')) {
-        assert_screen 'select-root-filesystem';
-        send_key 'alt-f';
-        send_key_until_needlematch 'filesystem-btrfs', 'down', 10, 3;
-        send_key 'ret';
-    }
-    else {
-        assert_and_click 'default-root-filesystem';
-        assert_and_click "filesystem-btrfs";
-    }
-    assert_screen "btrfs-selected";
-    send_key $cmd{next};
-}
-
-sub take_first_disk {
-    # create partitioning
-    send_key $cmd{createpartsetup};
-    assert_screen 'prepare-hard-disk';
-
-    wait_screen_change {
-        send_key 'alt-1';
-    };
-    send_key $cmd{next};
-
-    assert_screen 'use-entire-disk';
-    wait_screen_change {
-        send_key 'alt-e';
-    };
-    send_key $cmd{next};
-}
+use partition_setup 'take_first_disk';
 
 sub run {
-    if (is_storage_ng) {
```

Comment:
such explicit call would make more sense as for me. There is no logic related to `take_first_disk_storage_ng` inside `take_first_disk` . You just hiding call to one function inside another while they fully independent.  
Before this change I could clearly see that we have two separate flows one for storage_ng and another for all other cases

Reply:
Yeah, and it was intended to abstract from storage stack, but focus on what we are doing. If you see take_first_disk, you know that first disk is selected irregardless of the flow. And if you want to know the details you open the method. What we can do is to check if it storage-ng or not when we call the method and then send it as parameter, but then we will do 2 checks for the same thing.